tests/core/test_featurizer.py

```diff
@@ -137,16 +167,36 @@ def test_single_state_featurizer_correctly_encodes_state(
 
     f = SingleStateFeaturizer()
     f._default_feature_states[INTENT] = {"a": 0, "b": 1}
-    f._default_feature_states[ACTION_NAME] = {"e": 0, "d": 1}
     f._default_feature_states[ENTITIES] = {"c": 0}
+    f._default_feature_states[ACTION_NAME] = {"e": 0, "d": 1}
+    f._default_feature_states[SLOTS] = {"e_0": 0, "f_0": 1, "g_0": 2}
+    f._default_feature_states[ACTIVE_LOOP] = {"h": 0, "i": 1, "j": 2, "k": 3}
     encoded = f.encode_state(
         {
             "user": {"text": "a ball", "entities": ["c"]},
             "prev_action": {"action_name": "action_listen"},
+            "active_loop": {"name": "k"},
+            "slots": {"e": (1.0,)},
         },
         interpreter=kwargs["interpreter"],
     )
-    assert all([attribute in encoded for attribute in [TEXT, ENTITIES, ACTION_NAME]])
+    assert all(
+        [
+            attribute in encoded
+            for attribute in [TEXT, ENTITIES, ACTION_NAME, SLOTS, ACTIVE_LOOP]
+        ]
+    )
```

Comment:
what does it check?

Reply:
that all the features we expect to be encoded are encoded (to avoid the problem we had before with Forms and Slots)

Reply:
isn't the same as `assert list(encoded.keys()) == [TEXT, ENTITIES, ACTION_NAME, SLOTS, ACTIVE_LOOP]`